src/test/java/apoc/nodes/NodesTest.java

```diff
@@ -277,4 +276,191 @@ public void testRelType() {
         assertNull(db.execute("RETURN apoc.rel.type(null) AS type").columnAs("type").next());
     }
 
+    @Test
+    public void testMergeSelfRelationship() {
+        db.execute("MATCH (n) detach delete (n)");
+        db.execute("CREATE (a1:ALabel {name:'a1'})-[:KNOWS]->(b1:BLabel {name:'b1'})");
+
+        List<Label> label = asList(label("VirtualNode"), label("ALabel"), label("BLabel"));
+
+        TestUtil.testResult(db,
+                "MATCH (p:ALabel)-[r:KNOWS]->(c:BLabel) WITH p,c " +
+                        "CALL apoc.nodes.collapse([p,c],{mergeRels:true, selfRel: true}) yield from, rel, to " +
+                        "return from, rel, to", (r) -> {
+                    Map<String, Object> map = r.next();
+
+                    assertMerge(map,
+                            Util.map("name","b1", "count", 2), label, //FROM
+                            Util.map(), "KNOWS", //REL
```

Comment:
Non ho capito questa riga perchè?

Reply:
Vado a testare il tipo della relazione e le property

Reply:
E anche in tutti gli altri punti